htdocs/core/lib/functions.lib.php

```diff
@@ -5856,7 +5856,7 @@ function dolGetFirstLastname($firstname,$lastname,$nameorder=-1)
 	$ret='';
 	// If order not defined, we use the setup
 	if ($nameorder < 0) $nameorder=$conf->global->MAIN_FIRSTNAME_NAME_POSITION;
-	if ($nameorder && ((string) $nameorder != '2'))
+	if (empty($nameorder) && ((string) $nameorder != '2'))
```

Comment:
If you use here 
if empt(...) then the && ((string) $nameorder != '2') is useless.
Old code looks better to me. Can you check your PR ?

Reply:
This is values from the conf select MAIN_FIRSTNAME_NAME_POSITION
![image](https://user-images.githubusercontent.com/31647290/43629363-5346dd28-96fd-11e8-92fd-43e4c5b3870b.png)
So value '0' is used for Firstname+Lastname and not to Lastname+Firstname

Reply:
I saw in 6.0 version that something is missing in 7.0 :
```php
if ($nameorder < 0) $nameorder=empty($conf->global->MAIN_FIRSTNAME_NAME_POSITION);
```
With this test it will work correctly in 7.0